rpmlint/ReadelfParser.py

```diff
@@ -0,0 +1,126 @@
+import re
+import subprocess
+
+from itertools import dropwhile, takewhile
+
+class Issue:
+    """
+    A simple issues wrapper that later reported via add_info.
+    TODO: The class should be probably moved to filter.py.
+    """
+
+    def __init__(self, level, reason, *details):
+        self.level = level
+        self.reason = reason
+        # TODO: save details
+
+class ElfSection:
+    """
+    A simple wrapper representing one ELF section.
+    """
+    def __init__(self, name, size):
+        self.name = name
+        self.size = int(size, 16)
+
+"""
+Output example:
+
+There are 12 section headers, starting at offset 0x268:
+
+Section Headers:
+  [Nr] Name              Type            Address          Off    Size   ES Flg Lk Inf Al
+  [ 0]                   NULL            0000000000000000 000000 000000 00      0   0  0
+  [ 1] .text             PROGBITS        0000000000000000 000040 000015 00  AX  0   0  1
+  [ 2] .rela.text        RELA            0000000000000000 0001d8 000018 18   I  9   1  8
+  [ 3] .data             PROGBITS        0000000000000000 000055 000000 00  WA  0   0  1
+  [ 4] .bss              NOBITS          0000000000000000 000055 000000 00  WA  0   0  1
+  [ 5] .comment          PROGBITS        0000000000000000 000055 000041 01  MS  0   0  1
+  [ 6] .note.GNU-stack   PROGBITS        0000000000000000 000096 000000 00      0   0  1
+  [ 7] .eh_frame         PROGBITS        0000000000000000 000098 000038 00   A  0   0  8
+  [ 8] .rela.eh_frame    RELA            0000000000000000 0001f0 000018 18   I  9   7  8
+  [ 9] .symtab           SYMTAB          0000000000000000 0000d0 0000f0 18     10   8  8
+  [10] .strtab           STRTAB          0000000000000000 0001c0 000011 00      0   0  1
+  [11] .shstrtab         STRTAB          0000000000000000 000208 000059 00      0   0  1
+Key to Flags:
+  W (write), A (alloc), X (execute), M (merge), S (strings), I (info),
+  L (link order), O (extra OS processing required), G (group), T (TLS),
+  C (compressed), x (unknown), o (OS specific), E (exclude),
+  l (large), p (processor specific)
+"""
+
+class ElfSectionInfo:
+    """
+    Class contains information about ELF sections of an ELF file. The information
+    is get with the following command line: readelf -WS.
+    """
+
+    section_regex = re.compile(r'.*\] (?P<section>[^\s]*)\s*\w+\s*\w*\s*\w*\w*\s*(?P<size>\w*)')
```

Comment:
For later use could you please add explanation of this regexp above in a form of comment please?

Reply:
Ah you put it above, but this should be part of the class metadata not the module metadata anyway :)